daemon/api.go

```diff
@@ -116,124 +116,136 @@ var api = []*Command{
-		Path:    "/v2/system-info",
-		GuestOK: true,
-		GET:     sysInfo,
+		Path:       "/v2/system-info",
+		ReadAccess: openAccess{},
+		GET:        sysInfo,
 	}
 
 	appIconCmd = &Command{
-		Path:   "/v2/icons/{name}/icon",
-		UserOK: true,
-		GET:    appIconGet,
+		Path:       "/v2/icons/{name}/icon",
+		ReadAccess: openAccess{},
+		GET:        appIconGet,
 	}
 
 	findCmd = &Command{
-		Path:   "/v2/find",
-		UserOK: true,
-		GET:    searchStore,
+		Path:       "/v2/find",
+		ReadAccess: openAccess{},
+		GET:        searchStore,
 	}
 
 	snapsCmd = &Command{
-		Path:     "/v2/snaps",
-		UserOK:   true,
-		PolkitOK: "io.snapcraft.snapd.manage",
-		GET:      getSnapsInfo,
-		POST:     postSnaps,
+		Path:        "/v2/snaps",
+		ReadAccess:  openAccess{},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
+		GET:         getSnapsInfo,
+		POST:        postSnaps,
 	}
 
 	snapCmd = &Command{
-		Path:     "/v2/snaps/{name}",
-		UserOK:   true,
-		PolkitOK: "io.snapcraft.snapd.manage",
-		GET:      getSnapInfo,
-		POST:     postSnap,
+		Path:        "/v2/snaps/{name}",
+		ReadAccess:  openAccess{},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
+		GET:         getSnapInfo,
+		POST:        postSnap,
 	}
 
 	appsCmd = &Command{
-		Path:   "/v2/apps",
-		UserOK: true,
-		GET:    getAppsInfo,
-		POST:   postApps,
+		Path:        "/v2/apps",
+		ReadAccess:  openAccess{},
+		WriteAccess: authenticatedAccess{},
+		GET:         getAppsInfo,
+		POST:        postApps,
 	}
 
 	logsCmd = &Command{
-		Path:     "/v2/logs",
-		PolkitOK: "io.snapcraft.snapd.manage",
-		GET:      getLogs,
+		Path:       "/v2/logs",
+		ReadAccess: authenticatedAccess{Polkit: polkitActionManage},
+		GET:        getLogs,
 	}
 
 	snapConfCmd = &Command{
-		Path: "/v2/snaps/{name}/conf",
-		GET:  getSnapConf,
-		PUT:  setSnapConf,
+		Path:        "/v2/snaps/{name}/conf",
+		ReadAccess:  authenticatedAccess{},
+		WriteAccess: authenticatedAccess{},
+		GET:         getSnapConf,
+		PUT:         setSnapConf,
 	}
 
 	interfacesCmd = &Command{
-		Path:     "/v2/interfaces",
-		UserOK:   true,
-		PolkitOK: "io.snapcraft.snapd.manage-interfaces",
-		GET:      interfacesConnectionsMultiplexer,
-		POST:     changeInterfaces,
+		Path:        "/v2/interfaces",
+		ReadAccess:  openAccess{},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManageInterfaces},
+		GET:         interfacesConnectionsMultiplexer,
+		POST:        changeInterfaces,
 	}
 
 	stateChangeCmd = &Command{
-		Path:     "/v2/changes/{id}",
-		UserOK:   true,
-		PolkitOK: "io.snapcraft.snapd.manage",
-		GET:      getChange,
-		POST:     abortChange,
+		Path:        "/v2/changes/{id}",
+		ReadAccess:  openAccess{},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
+		GET:         getChange,
+		POST:        abortChange,
 	}
 
 	stateChangesCmd = &Command{
-		Path:   "/v2/changes",
-		UserOK: true,
-		GET:    getChanges,
+		Path:       "/v2/changes",
+		ReadAccess: openAccess{},
```

Comment:
I have a small conceptual problem with this change. Before it was clear that it had to be a local access that was allowed. Now we call it `openAccess` and that's effectively the same but once we grow any kind of remote access existing distinction will be lost. It also bothers me somewhat that `open` implies only the classic world and secretly (though that's not something I disagree with) excludes access from snaps. 

How about we call `openAccess` `localSystemAccess` instead. This way we clearly indicate this is for local interactions only, we can add `remoteSystemAccess` later and we already have `snapAccess` that is both local (by definition) and non-system.

Reply:
I chose the names to match what is used in the REST API documentation, which I think holds some value.  If we are going to change this, it should be something that would be understandable in the docs.

We can always rename things when TCP access to the API is a thing.

Reply:
I don't think using the term "local" in the access name will help at this point. It's very likely also that the first form of remote access will be via the store, not accessing the snapd own socket over http. Indeed so far we tend to use "remote" to mean the store.